lib/server.js

```diff
@@ -38,31 +38,30 @@ exports.run = (req, res, fn) =>
 // multiple calls to `json` work as expected
 const rawBodyMap = new WeakMap()
 
-const returnJSON = (resolve, reject, str) => {
+const parseJSON = str => {
   try {
-    resolve(JSON.parse(str))
+    return JSON.parse(str)
   } catch (err) {
-    reject(createError(400, 'Invalid JSON', err))
+    throw createError(400, 'Invalid JSON', err)
   }
 }
 
-exports.json = (req, {limit = '1mb'} = {}) => new Promise((resolve, reject) => {
+exports.json = (req, {limit = '1mb'} = {}) => resolve().then(() => {
```

Comment:
@albinekb it destructed [here](https://github.com/zeit/micro/blob/master/lib/server.js#L10). Actually `const Promise` can be removed for now.